thetis/solver2d.py

```diff
@@ -398,6 +435,68 @@ def get_tracer_timestepper(self, integrator):
             kwargs['theta'] = self.options.timestepper_options.implicitness_theta
         return integrator(*args, **kwargs)
 
+    def get_sediment_timestepper(self, integrator):
+        """
+        Gets sediment timestepper object with appropriate parameters
+        """
+        uv, elev = self.fields.solution_2d.split()
+        fields = {
+            'elev_2d': elev,
+            'uv_2d': uv,
+            'diffusivity_h': self.options.horizontal_diffusivity,
+            'source': self.options.sediment_model_options.sediment_ero,
+            'depth_integrated_source': self.options.sediment_model_options.sediment_depth_integ_ero,
+            'sink': self.options.sediment_model_options.sediment_depo,
+            'depth_integrated_sink': self.options.sediment_model_options.sediment_depth_integ_depo,
```

Comment:
Or, as the sediment equation is quite specific, it could have a specific source term which actually defines the erosion/deposition expressions (similarly to the turbulence equations). Same for the exner equation.

Reply:
following your advice we now use a get_erosion_term function to get these expressions